lib/spanner/config/parser.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Spanner.Config.Parser do
+
+  @moduledoc """
+  Functions for parsing YAML config files
+  """
+
+  @doc """
+  Reads a YAML file from 'path'. The underlying lib, yamerl, throws when
+  errors occur. We catch those and return tuple containing the atom ':error'
+  and the list error messages.
+  """
+  def read_from_file(path) do
+    try do
+      yaml = YamlElixir.read_from_file(path)
+      # We should never get an empty map back from 'YamlElixir.read_from_file/1'.
+      # If we do, we return an error.
+      if length(Map.values(yaml)) == 0 do
+        {:error, ["Empty map returned. Make sure there are no errors in your YAML."]}
+      else
+        {:ok, yaml}
+      end
+    catch
+      {:yamerl_exception, errors} ->
+        {:error, Enum.map(errors, &format_errors/1)}
+    end
+  end
+
+  @doc """
+  Reads YAML from a string. The underlying lib, yamerl, throws when
+  errors occur. We catch those and return tuple containing the atom ':error'
+  and the list error messages.
+  """
+  def read_from_string(str) do
```

Comment:
It gets used in relay when we read configs from the `.cog` file. https://github.com/operable/relay/pull/44/files#diff-1261558923958c8980bb78462a004da1R64